Fixes #486

larswik opened the pull request:
Fixes #486. Also saw that IfcFillElementTypeEnum lacked description so I added that as well here.

larswik committed:
Fixes #486

Also saw that IfcFillElementTypeEnum lacked description so I added that as well here.

pjanck commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElementType/Documentation.md:
```diff
@@ -0,0 +1,10 @@
+The element type _IfcArchElementType_ defines commonly shared information for occurrences of arch elements. Arch elements are building elements which represents unitary curved structures.The set of shared information may include:
```
```suggestion
The element type _IfcArchElementType_ defines commonly shared information for occurrences of arch elements. Arch elements are building elements which represents unitary curved structures. The set of shared information may include:
```

pjanck commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElementType/Documentation.md:
```diff
@@ -0,0 +1,10 @@
+
+It is used to define an arch element specification that is common to all occurrences of that arch element type. Arch element types may be exchanged without being already assigned to occurrences.
+
+Occurrences of the _IfcFillElementType_ within building models are represented by instances of _IfcFillElement_.
```
*Fill*?

SergejMuhic replied:
```suggestion
Occurrences of the _IfcArchElementType_ within building models are represented by instances of _IfcArchElement_.
```

SergejMuhic commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Types/IfcArchElementTypeEnum/Documentation.md:
```diff
@@ -0,0 +1 @@
+This enumeration defines the different types of elements an _IfcArchElement_ or _IfcArchElementType_ object can fulfill.
```
Feel free to reject. Just thought it sounds better and fits better with the relationship.
```suggestion
This enumeration defines the different types of elements an _IfcArchElement_ or _IfcArchElementType_ object can be defined as.
```

SergejMuhic commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Types/IfcFillElementTypeEnum/Documentation.md:
```diff
@@ -0,0 +1 @@
+This enumeration defines the different types of elements an _IfcFillElement_ or _IfcFillElementType_ object can fulfill.
```
```suggestion
This enumeration defines the different types of elements an _IfcFillElement_ or _IfcFillElementType_ object can be defined as.
```

SergejMuhic commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElementType/DocEntity.xml:
```diff
@@ -0,0 +1,16 @@
+	<WhereRules>
+		<DocWhereRule Name="CorrectPredefinedType" UniqueId="9e780a13-680e-4fce-939a-948f87163eee">
+			<Documentation>The inherited attribute _ElementType_ shall be provided, if the _PredefinedType_ is set to USERDEFINED.</Documentation>
+			<Expression>(PredefinedType &lt;&gt; IfcArchElementTypeEnum.USERDEFINED) OR
```
I think I should turn off this html encoding. Really looks bad. 😄

SergejMuhic commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/Documentation.md:
```diff
@@ -0,0 +1 @@
+A unitary curved structure.
```
I like Meriam Webster: "a typically curved structural member spanning an opening and serving as a support"

https://www.merriam-webster.com/dictionary/arch

larswik replied:
I agree but I just used the definition from IfcElementAssembly/ARCH and added "unitary". Should we change there as well then?

SergejMuhic replied:
I would not dare. That is bSI business.

larswik replied:
> I like Meriam Webster: "a typically curved structural member spanning an opening and serving as a support"
> 
> https://www.merriam-webster.com/dictionary/arch

Please check now.

SergejMuhic commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/DocEntity.xml:
```diff
@@ -0,0 +1,24 @@
+<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcArchElement" UniqueId="497395c2-219d-4d42-9eaf-1a8c3f3967cd" EntityFlags="32">
+	<Attributes>
+		<DocAttribute Name="PredefinedType" UniqueId="140250ca-4d21-4dca-9f29-7b99e1c6857b" DefinedType="IfcArchElementTypeEnum" AttributeFlags="1">
+			<Documentation>Predefined generic type for an arch element that is specified in an enumeration. There may be a property set given specifically for the predefined types.
```
Did we not have a more generic common description that we used? This, seems to me, is not it. 🤔

larswik replied:
There are a few...A generic one I found is: "Specifies the type for which the value is selected from a predefined type enumeration. This type may associate additional specific property sets".

SergejMuhic replied:
This wording "Predefined generic type" bothers me a bit. What is generic about the predefined type? 🤔

larswik replied:
> This wording "Predefined generic type" bothers me a bit. What is generic about the predefined type? 🤔

Please check now.

larswik committed:
Update IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElementType/Documentation.md

Co-authored-by: Stefan Jaud <59165496+pjanck@users.noreply.github.com>

larswik committed:
Update IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElementType/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Update IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Types/IfcArchElementTypeEnum/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Update IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Types/IfcFillElementTypeEnum/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

pjanck reviewed (approved):
lftm

larswik committed:
Fixing according to review comments

- Use other definition for PredefinedType attribute
- Modified definition (used Merriem-Webster as proposed)

anborr commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElementType/Documentation.md:
```diff
@@ -0,0 +1,10 @@
+The element type _IfcArchElementType_ defines commonly shared information for occurrences of arch elements. Arch elements are building elements which represents unitary curved structures. The set of shared information may include:
```
```suggestion
The element type _IfcArchElementType_ defines commonly shared information for occurrences of arch elements. Arch elements are building elements which represent unitary curved structures. The set of shared information may include:
```

SergejMuhic commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/Documentation.md:
```diff
@@ -0,0 +1 @@
+A typically curved structural member spanning an opening and serving as a support
```
```suggestion
A typically curved structural member spanning an opening and serving as a support.
```

SergejMuhic committed:
Update IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElementType/Documentation.md

Co-authored-by: anborr <73635642+anborr@users.noreply.github.com>

SergejMuhic committed:
Update IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedBldgElements/Entities/IfcArchElement/Documentation.md

SergejMuhic commented:
WIll merge after schema generation with commit hash has been implemented.
